lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+
+    $verb = ($options['audit']) ? 'audited' : 'processed';
+    $this->logSection(sprintf('Successfully %s %d digital objects.', $verb, $count));
+  }
+
+  /**
+   * Make sure argument and option values are valid
+   *
+   * @param array $arguments  array of sfCommandArgument instances
+   * @param array $options  array of sfCommandOption instances
+   *
+   * @return void
+   */
+  protected function checkArgumentsAndOptions($arguments, $options)
+  {
+    // Make sure DIP directory exists
+    if (!is_dir($arguments['dip']))
+    {
+      throw new sfException('You must specify a DIP directory');
+    }
+
+    // If undo log directory specified, make sure it's a valid directory 
+    if (!empty($options['undo-log-dir']) && !is_dir($options['undo-log-dir']))
+    {
+      throw new sfException('Undo log directory does not exist.');
+    }
+  }
+
+  /**
+   * Return file pointer resource for the first CSV file found within the DIP's
+   * objects directory. CSV file's extension must be lower-case.
+   *
+   * @return resource  File pointer resource for CSV file
+   */
+  protected function openFirstCsvFile()
+  {
+    // Find all CSV files in the objects directory
+    $csvFiles = sfFinder::type('file')->name('*.csv')->in($this->dipDir .'/objects');
+
+    // Attempt to open the first CSV file found
+    if (0 == count($csvFiles) || false === $fh = fopen($csvFiles[0], 'rb'))
+    {
+      throw new sfException('You must specify a DIP that contains a CSV file');
+    }
+
+    return $fh;
+  }
+
+  /**
+   * Determine CSV column order and which information object attribute (slug or
+   * identifier) will be used as unique value to specify information objects to
+   * timport digital objects to. Read CSV data, creating an array of information
+   * object unique values and the original filenames of digital objects that
+   * should be imported and associated with these information objects.
+   *
+   * @param resource $fh  File pointer resource for CSV file
+   * @param string $objectsPath  Path to DIP object files
+   *
+   * @return array  Hash with hash key being unique value and hash value(s)
+   *                being original object filenames
+   */
+  protected function parseCsvData($fh, $objectsPath)
+  {
+    // Create a lookup table to match original file basename (UUID and
+    // extension removed) with original object filename (example:
+    // 979c4458-21f3-11e1-a4bd-001d09282b9d-foo.tif)
+    $filenames = $this->createFilenameLookup($objectsPath);
+
+    // Determine column order and what type of unique value will be used to find
+    // information objects (slug or identifier)
+    $this->processCsvHeaderRow($fh);
+
+    $digitalObjects = array();
+
+    // Build hash on information_object key, with value being an array if information
+    // object has multiple digital objects attached
+    while ($row = fgetcsv($fh, 1000))
+    {
+      $filename = $this->getRowColumnValue('filename', $row);
```

Comment:
Could we get the filename column index only once instead of making am array_search() for each row?